src/node/miner.cpp

```diff
@@ -262,23 +262,6 @@ int BlockAssembler::UpdatePackagesForAdded(const CTxMemPool::setEntries& already
     return nDescendantsUpdated;
 }
 
-// Skip entries in mapTx that are already in a block or are present
-// in mapModifiedTx (which implies that the mapTx ancestor state is
-// stale due to ancestor inclusion in the block)
-// Also skip transactions that we've already failed to add. This can happen if
-// we consider a transaction in mapModifiedTx and it fails: we can then
-// potentially consider it again while walking mapTx.  It's currently
-// guaranteed to fail again, but as a belt-and-suspenders check we put it in
-// failedTx and avoid re-evaluation, since the re-evaluation would be using
-// cached size/sigops/fee values that are not actually correct.
-bool BlockAssembler::SkipMapTxEntry(CTxMemPool::txiter it, indexed_modified_transaction_set& mapModifiedTx, CTxMemPool::setEntries& failedTx)
```

Comment:
in the first commit:

Any reason to not use the same approach from the second commit and make this function `static`? I understand that this would require passing the mempool solely to check the lock annotation, but I think this makes the diff smaller, keeps the structure of the file, and avoids making a large function even larger by inlining.

If there are no other reasons I'd prefer to keep this as-is for now and leave restructuring to changes that actually need it?

Reply:
When I was looking at it, this function just seemed so intertwined with the rest of `addPackageTxs`. Both `mapModifiedTx` and `failedTx` are local variables to `addPackageTxs` and without the function abstraction we can also drop the `AssertLockHeld`. I agree that `addPackageTxs` is quite large and we could break it down into pieces where appropriate but I'm not sure that this is the right place.

If we were to make it static we'd have to pass in the mempool and inBlock.